api-module-library/frontify/api.js

```diff
@@ -52,48 +52,148 @@ class Api extends OAuth2Requester {
         return response.data.brand.libraries.items;
     }
 
     async listProjectAssets(query) {
-        const ql = `query ProjectAssets { workspaceProject(id: "${query.projectId}") { assets { items { id title description }}}}`;
+        const ql = `query ProjectAssets {
+                      workspaceProject(id: "${query.projectId}") {
+                        assets {
+                          items {
+                            id
+                            title
+                            description
+                            __typename
+                          }
+                        }
+                      }
+                    }`;
+
         const response = await this._post(this.buildRequestOptions(ql));
         return {
             assets: response.data.workspaceProject.assets.items,
         };
     }
 
+    async listProjectFolders(query) {
+        const ql = `query ProjectFolders {
+                      workspaceProject(id: "${query.projectId}") {
+                        browse {
+                          folders {
+                            items {
+                              id
+                              name
+                              __typename
+                            }
+                          }
+                        }
+                      }
+                    }`;
+
+        const response = await this._post(this.buildRequestOptions(ql));
+        return {
+            folders: response.data.workspaceProject.browse.folders.items
+        };
+    }
+
     async listLibraryAssets(query) {
-        const ql = `query LibraryAssets { library(id: "${query.libraryId}") { assets { items { id title description }}}}`;
+        const ql = `query LibraryAssets {
+                      library(id: "${query.libraryId}") {
+                        assets {
+                          items {
+                            id
+                            title
+                            description
+                            __typename
+                          }
+                        }
+                      }
+                    }`;
+
         const response = await this._post(this.buildRequestOptions(ql));
         return {
             assets: response.data.library.assets.items,
         };
     }
+
+    async listLibraryFolders(query) {
+        const ql = `query LibraryFolders {
+                      library(id: "${query.libraryId}") {
+                        browse {
+                          folders {
+                            items {
+                              id
+                              name
+                              __typename
+                            }
+                          }
+                        }
+                      }
+                    }`;
+
+        const response = await this._post(this.buildRequestOptions(ql));
+        return {
+            folders: response.data.library.browse.folders.items
```

Comment:
@roboli What happens if there are no results, the object still goes down to items?

Reply:
Returns an empty array coming from Frontify GraphQL api:

```
{
  "data": {
    "library": {
      "browse": {
        "folders": {
          "items": []
        }
      }
    }
  },
  "extensions": {
    "complexityScore": 0
  }
}
```

Reply:
Only exception is when trying to get folders from non existent Library or Project (using a bad ID for example). In that case method should throw an error :thinking: 

This from GraphQL api:

```
{
  "errors": [
    {
      "message": "Internal server error",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "library"
      ],
      "extensions": {
        "category": "internal"
      }
    }
  ],
  "data": {
    "library": null
  },
  "extensions": {
    "complexityScore": 0
  }
}
```

Reply:
Hmmmm nope, request returns `200` with the error above as the result... I'll work on making methods throw an error/exception if that's the case @leofmds